src/app/components/AutoComplete.svelte

```diff
@@ -0,0 +1,188 @@
+<script lang="ts">
+  // TODO: the following interface should be removed once #33 is done
+  interface dataObj {
+    bcp47Tag: string;
+    language: string;
+  }
+  enum keyboardKey {
+    Up = "ArrowUp",
+    Down = "ArrowDown",
+    Enter = "Enter",
+  }
+  export let label = "";
+  export let subtext = "";
+  // Testing Data
+  export let results: dataObj[] = [];
+  // To store Filted array
+  let filetred: dataObj[] = [];
+  // To store selected language
+  let selected: string = "";
+  // Toggle to show search list
+  let show = false;
+  // Index of Previous focus element
+  let prevIndex = -1;
+  // Index of focus element
+  let index = -1;
+
+  // Input to search
+  function onChange(event: any) {
+    show = true;
+    filetred = results.filter((item: any) => {
+      let a = new RegExp("^" + event.target.value.toUpperCase());
+      return a.test(item.language.toUpperCase());
+    });
+  }
+
+  function closeSuggestion() {
+    show = false;
+  }
+
+  // If anywhere on the screen is clicked
+  // Derived from https://github.com/rster2002/svelte-outside-click
+  function clickOutside(node: HTMLUListElement, onEventFunction: Function) {
+    const handleClick = (event: MouseEvent) => {
+      const path = event.composedPath();
+      if (!path.includes(node)) {
+        onEventFunction();
+      }
+    };
+    document.addEventListener("click", handleClick);
+    return {
+      destroy() {
+        document.removeEventListener("click", handleClick);
+      },
+    };
+  }
+
+  // On select item in list
+  function selectedList(data: dataObj) {
+    show = false;
+    selected = data.language;
+    subtext = data.bcp47Tag;
+  }
+
+  const mod = (a: number, n: number) => {
+    return a - n * Math.floor(a / n);
+  };
+
+  // Up/Down arraow
+  function handleKeydown({ key }: KeyboardEvent) {
+    if (
+      key !== keyboardKey.Down &&
+      key !== keyboardKey.Up &&
+      key !== keyboardKey.Enter
+    )
+      return;
+    if (key == keyboardKey.Down) {
+      prevIndex = index;
+      index += 1;
+    } else if (key == keyboardKey.Up) {
+      if (index == -1) {
+        index = -1;
+      } else {
+        prevIndex = index;
+        index -= 1;
+      }
+    } else if (key == keyboardKey.Enter) {
+      selected = filetred[index].language;
+      subtext = filetred[index].bcp47Tag;
+      show = false;
+    }
```

Comment:
Remove the need for the first conditional by handling it in the `else` clause:

```suggestion
    if (key === keyboardKey.Down) {
      prevIndex = index;
      index += 1;
    } else if (key === keyboardKey.Up) {
      if (index == -1) {
        index = -1;
      } else {
        prevIndex = index;
        index -= 1;
      }
    } else if (key === keyboardKey.Enter) {
      selected = filetred[index].language;
      subtext = filetred[index].bcp47Tag;
      show = false;
    } else {
      /* Not a key we care about */
      return;
    }
```

You could even use a `switch`/`case`/`default` if you're so bold! 👀

Reply:
omg I love this defector and it is way my readable then my previous code!!!